### PR TITLE
[ZUM-29491] Fix sighup race conditions

### PR DESCRIPTION
**Pull request by jvineet**
resolves:
ZUM-29491

running in test and prod nomad

### CODE REVIEW

File: prometheus_pgbouncer_exporter/cli.py [hunk 1]
```diff
@@ -16,23 +16,24 @@ def read_config_file(config_file):
...
-        logging.getLogger().fatal("The config file {file} is invalid: {error}".format(file=args.config, error=str(error)))
+        logging.getLogger().fatal("The config file {file} is invalid: {error}".format(file=config_file, error=str(error)))
         raise Exception('Invalid config file')
 
     return config
 
 
 def main():
     shutdown = False
+    sighup_received_count = {'val': 0}
```
Comment (dbachelder) [inline prometheus_pgbouncer_exporter/cli.py]: why is this a dict?

File: prometheus_pgbouncer_exporter/cli.py [hunk 2]
```diff
@@ -51,9 +52,9 @@ def main():
 
     # Register signal handler
     def _on_sighup(signal, frame):
-        logging.getLogger().info("Received SIGHUP - Reloading pgbouncer hosts")
-        config = read_config_file(args.config)
-        pgbcollector.update(config.getPgbouncers())
+        sighup_received_count['val'] += 1
```
Comment (dbachelder) [inline prometheus_pgbouncer_exporter/cli.py]: comment about why we are doing things this way

**Review by dbachelder — APPROVED** [review]
lgtm

File: prometheus_pgbouncer_exporter/cli.py (hunk 1 above)
Comment (hlawrenz) [reply prometheus_pgbouncer_exporter/cli.py]: https://stackoverflow.com/questions/3190706/nonlocal-keyword-in-python-2-x

This has some alternatives although this hack is the top answer fwiw. Personally I don't like the function attr solution as it's even less obvious. The other suggestion is to make a class or object to hold the nonlocal.

Comment (hlawrenz) [reply prometheus_pgbouncer_exporter/cli.py]: I didn't finish my thought :) I'd just go with the dict and add a comment about why you're doing it that way.

Comment (jvineet) [reply prometheus_pgbouncer_exporter/cli.py]: Just found out this project only supports python>3, so we can use `nonlocal` in the inner function here.

**Review by hlawrenz — APPROVED** [review]
lgtm

**Pull request merged**